version.props

```diff
@@ -5,6 +5,7 @@
     <PackageVersion Condition="'$(IsFinalBuild)' == 'true' AND '$(VersionSuffix)' == 'rtm' ">$(VersionPrefix)</PackageVersion>
     <PackageVersion Condition="'$(IsFinalBuild)' == 'true' AND '$(VersionSuffix)' != 'rtm' ">$(VersionPrefix)-$(VersionSuffix)-final</PackageVersion>
     <BuildNumber Condition="'$(BuildNumber)' == ''">t000</BuildNumber>
-    <VersionSuffix Condition="'$(VersionSuffix)' != '' And '$(BuildNumber)' != ''">$(VersionSuffix)-$(BuildNumber)</VersionSuffix>
+    <SanitizedBuildNumber>$([System.Text.RegularExpressions.Regex]::Replace('$(BuildNumber)', '[^\w-]', '-'))</SanitizedBuildNumber>
```

Comment:
Any reason this shouldn't just replace the original value of BuildNumber instead of making a new property? BuildNumber is used in other places

Reply:
This doesn't work no - `dotnet msbuild /p:BuildNumber=branch/name`

Reply:
So you are planning to pass `/p:BuildNumber=%teamcity.build.branch%/%teamcity.build.number%` on build? That might cause some problems because you would need to update every usage of `$(BuildNumber)` to `$(SanitizedBuildNumber)`. Maybe it would be easier to set a new property instead of overriding an existing one. For example, we could do

args: `/p:FeatureBranchVersionSuffix=%teamcity.build.branch%`
version.props:
```xml
<VersionSuffix>preview1</VersionSuffix>
<VersionSuffix Condition=" '$(p:FeatureBranchVersionSuffix)' != '' ">$(VersionSuffix)-$([System.Text.RegularExpressions.Regex]::Replace('$(p:FeatureBranchVersionSuffix)', '[^\w-]', '-'))</VersionSuffix>
```

Reply:
Sure, that works